grammar.js

```diff
@@ -611,7 +615,7 @@ module.exports = grammar(C, {
       optional('::'),
       'delete',
       optional(seq('[', ']')),
-      $._expression
+      $.identifier
```

Comment:
I don't believe you can delete any type of expression; I'm pretty sure it's limited to just identifiers.